src/System.Threading.Tasks.Extensions/tests/AsyncValueTaskMethodBuilderTests.cs

```diff
@@ -0,0 +1,131 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Threading.Tasks.Tests
+{
+    public class AsyncValueTaskMethodBuilderTests
+    {
+        [Fact]
+        public void Create_ReturnsDefaultInstance()
+        {
+            AsyncValueTaskMethodBuilder<int> b = ValueTask<int>.CreateAsyncMethodBuilder();
+            Assert.Equal(default(AsyncValueTaskMethodBuilder<int>), b);
+        }
+
+        [Fact]
+        public void SetResult_BeforeAccessTask_ValueTaskContainsValue()
+        {
+            AsyncValueTaskMethodBuilder<int> b = ValueTask<int>.CreateAsyncMethodBuilder();
+            b.SetResult(42);
+            ValueTask<int> vt = b.Task;
+            Assert.True(vt.IsCompletedSuccessfully);
+            Assert.NotSame(vt.AsTask(), vt.AsTask()); // will be different if completed synchronously
+            Assert.Equal(42, vt.Result);
+        }
+
+        [Fact]
+        public void SetResult_AfterAccessTask_ValueTaskContainsValue()
+        {
+            AsyncValueTaskMethodBuilder<int> b = ValueTask<int>.CreateAsyncMethodBuilder();
+            ValueTask<int> vt = b.Task;
+            b.SetResult(42);
+            Assert.True(vt.IsCompletedSuccessfully);
+            Assert.Same(vt.AsTask(), vt.AsTask()); // will be safe if completed asynchronously
+            Assert.Equal(42, vt.Result);
+        }
+
+        [Fact]
+        public void SetException_BeforeAccessTask_ValueTaskContainsValue()
```

Comment:
Should the suffix be something like `_IsFaulted` rather than `_ValueTaskContainsValue`? Same question for test below.